src/interface/swarm_container.cpp

```diff
@@ -75,15 +76,41 @@ void SwarmContainer::Remove(const std::string &label) {
 
 void SwarmContainer::SendBoundaryBuffers() {}
 
-void SwarmContainer::SetupPersistentMPI() {}
+void SwarmContainer::SetupPersistentMPI() {
+  for (auto &s : swarmVector_) {
+    s->SetupPersistentMPI();
+  }
+}
 
 bool SwarmContainer::ReceiveBoundaryBuffers() { return true; }
 
 void SwarmContainer::ReceiveAndSetBoundariesWithWait() {}
 
 void SwarmContainer::SetBoundaries() {}
```

Comment:
stylistic question: Why did you not reuse these functions (so that the approach is similar to the other containers) but implemented them in a separate interface?

Reply:
`SwarmContainer`s are still due to go away... I think `Swarm`s should just go into `MeshBlockData`, but I think there are some outstanding issues that need to be thought through here, like if we want to communicate particles several times but variables only once. My understanding is this probably means extending `MeshBlockData` somewhat, and I think that should be a subsequent PR.